clang/include/clang/Sema/Sema.h

```diff
@@ -3391,7 +3391,7 @@ class Sema final : public SemaBase {
       bool ConstexprSupported, bool CLinkageMayDiffer);
 
   /// type checking declaration initializers (C99 6.7.8)
-  bool CheckForConstantInitializer(Expr *e, QualType t);
+  bool CheckForConstantInitializer(Expr *Init, unsigned DiagID);
```

Comment:
you could consider adding a default for DiagID as there's an "obvious" one used in most places. Up to you.

Reply:
Thanks for reviewing, I'll add default `DiagID` to `diag::err_init_element_not_constant`.